torch/csrc/autograd/engine.cpp

```diff
@@ -73,14 +74,33 @@ auto Engine::backward(const variable_list& variables,
         did_leaf_backward = true;
       }
     } else {
-      creators.push_back(var->creator.get());
-      if (var->creator->requires_grad) {
-        GradBuffer buf(var->creator->num_outputs);
-        buf.addGrad(var->output_nr, Variable::of(std::move(grad)));
-        ready.emplace_front(var->creator, std::move(buf));
+      auto& creator = var->creator;
+      auto it = creator_grad.find(creator);
```

Comment:
You can probably combine the two cases a bit since unique_ptr is default constructible by doing something like:

```
if (var->requires_grad) {
 auto& buf = creator_grad[creator]
 if (!buf) {
   buf.reset(new ...)
 }
 buf->addGrad(...)
}
```